src/containerapp/azext_containerapp/_utils.py

```diff
@@ -282,136 +285,161 @@ def get_pack_exec_path():
         result["reason"] = "The image is already up to date."
     return result
 
 
-def get_current_mariner_tags() -> list(OryxMarinerRunImgTagProperty):
-    r = requests.get("https://mcr.microsoft.com/v2/oryx/builder/tags/list", timeout=30)
-    tags = r.json()
-    tag_list = {}
-    # only keep entries that contain keyword "mariner"
-    tags = [tag for tag in tags["tags"] if "mariner" in tag]
+def get_oryx_run_image_tags() -> dict:
+    result = {}
+    result.update(_get_oryx_run_image_tags("https://mcr.microsoft.com/v2/oryx/dotnetcore/tags/list", parse_oryx_run_image))
+    result.update(_get_oryx_run_image_tags("https://mcr.microsoft.com/v2/oryx/node/tags/list", parse_oryx_run_image))
+    result.update(_get_oryx_run_image_tags("https://mcr.microsoft.com/v2/oryx/python/tags/list", parse_oryx_run_image))
+
+    # Return the merged result of all Oryx-supported platform tags
+    return result
+
+
+def _get_oryx_run_image_tags(tags_list_url, tag_parse_func) -> dict:
+    r = requests.get(tags_list_url, timeout=30)
+    response = r.json()
+    image_repository = response["name"]  # e.g., "oryx/dotnetcore" for "https://mcr.microsoft.com/v2/oryx/dotnetcore/tags/list"
+    tag_dict = {}
+
+    tags = list(response["tags"])
     for tag in tags:
-        tag_obj = parse_oryx_mariner_tag(tag)
+        tag_obj = tag_parse_func(image_repository, tag)
         if tag_obj:
-            major_minor_ver = str(tag_obj["version"].major) + "." + str(tag_obj["version"].minor)
             support = tag_obj["support"]
             framework = tag_obj["framework"]
-            mariner_ver = tag_obj["marinerVersion"]
-            if framework not in tag_list:
-                tag_list[framework] = {major_minor_ver: {support: {mariner_ver: [tag_obj]}}}
-            elif major_minor_ver not in tag_list[framework]:
-                tag_list[framework][major_minor_ver] = {support: {mariner_ver: [tag_obj]}}
-            elif support not in tag_list[framework][major_minor_ver]:
-                tag_list[framework][major_minor_ver][support] = {mariner_ver: [tag_obj]}
-            elif mariner_ver not in tag_list[framework][major_minor_ver][support]:
-                tag_list[framework][major_minor_ver][support][mariner_ver] = [tag_obj]
+            os_prop = tag_obj["os"]
+            version_key = _get_oryx_run_image_version_key(tag_obj, framework)
+            if framework not in tag_dict:
+                tag_dict[framework] = {version_key: {support: {os_prop: [tag_obj]}}}
+            elif version_key not in tag_dict[framework]:
+                tag_dict[framework][version_key] = {support: {os_prop: [tag_obj]}}
+            elif support not in tag_dict[framework][version_key]:
+                tag_dict[framework][version_key][support] = {os_prop: [tag_obj]}
+            elif os_prop not in tag_dict[framework][version_key][support]:
+                tag_dict[framework][version_key][support][os_prop] = [tag_obj]
             else:
```

Comment:
will the response always follow the expected format?

Reply:
I'm optimistic that the MCR team will do their part to avoid breaking changes within their `v2` API that we're calling here, but since we're not the owners of this registry (or the API), I'm not personally able to make any guarantees that this expected format will _always_ remain the same.

Reply:
Yes, I confirm it's fine. This scenario should never happen, and if the MCR team introduces a breaking change that would break this convention, then it would be the kind of scenario that would deserve an exception to be raised. :)